Automattic/wp-calypso
# Themes - Fix going "back" to themes showcase and losing my place in browsing themes.

#### Changes proposed in this Pull Request

* Add a `themesBookmark` section to the data store to allow for users to consistently return to their place in the browsing the showcase.

#### How this works
When a user clicks on a Theme card, the theme id is stored as the bookmark.  

When the `ThemesShowcase` mounts, it checks to see if the bookmark is set and scrolls to the card if it exists on the screen.

The theme is found by a React `ref` that is created in the `ThemesShowcase` and passed down child components until it gets to the `ThemeList`.  Here, the `ThemeList` decides whether or not to continue passing the `ref` down to the `Theme` component based on the `id` of the theme and the bookmark.

#### Testing instructions



* Run this PR locally.
* Navigate to themes section.
* Open "More Themes", scroll down, and select "Info" for some theme.
* Hit the "back" button.
* Verify we are back at the theme we selected.
* Hit "Info" again (for the same or a different theme), and hit the "Back" button again.  Verify we are still led back to the theme we had selected.
* Navigate to another section of calypso, such as "Plan" or "My Site".
* Hit "Themes" again, verify that we are looking at the theme we last viewed info for.
* Hit "Themes" again, verify we return to the top of the themes page.

Fixes #38182
